File: 백준/Silver/1912. 연속합/연속합.java
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		int[] A = new int[n+1];
		int[] D = new int[n+1];
		String[] str = br.readLine().split(" ");
		int max = -1000;
		for(int i=1; i<=n; i++) {
			A[i] = Integer.parseInt(str[i-1]);
			D[i] = Math.max(D[i-1] + A[i], A[i]);
			max = (D[i]>max)?D[i]:max;
		}
		System.out.print(max);
	}
}

File: 백준/Silver/1912. 연속합/연속합.py
n = int(input())
a = list(map(int, input().split()))
d = [-1001]*n
max = d[0] = a[0]
for i in range(1, n) :
    if d[i-1] + a[i] > a[i] :
        d[i] = d[i-1] + a[i]
    else :
        d[i] = a[i]
    if d[i] > max :
        max = d[i]
print(max)